lib/spack/spack/util/path.py

```diff
@@ -270,9 +271,9 @@ def repl(match):
     return re.sub(r"(\$\w+\b|\$\{\w+\})", repl, path)
 
 
-def substitute_path_variables(path):
+def substitute_path_variables(path, allow_env=True):
```

Comment:
What happens if you activate an environment that changes the location of the default environment list? I have a feeling Spack will explode if you do that, we should add some guardrails for it.

Reply:
Agreed.